readme.md

```diff
@@ -1,15 +1,95 @@
+
+## Implementation Notes
+Three update functions are responsible for collecting the necessary data in terms of air quality, areas, and stations, located in respectively labelled controller files in the repository:  ``` src/controllers/air-quality.controller.ts ```,  ``` src/controllers/area.controller.ts ```, and  ``` src/controller/station.controller.ts ```. The data is collected from the [MET air quality API](https://api.met.no/weatherapi/airqualityforecast/0.1/documentation). An air quality API from Telenor and NTNU was under development during the course project, however it did not provide all the information required for the application concept. Instead of fetching only certain items from either API, it was decided that the data would continue to be collected from the MET API, until the other API was fully featured. The base URL for the API used for these updating functions is located in ``` src/constants.ts ```. 
+
+The current updating solution relies on an Amazon Lambda function to hit the corresponding API endpoints to trigger the update. Internal timed tasks would not work in a monolith architecture when scaling occurs, as the action would be duplicated .
+
+The user account system relies on _Amazon Web Services'_ ([AWS](https://aws.amazon.com/)) _Identity & Access Management_ ([IAM](https://aws.amazon.com/iam/)) system by using [_Amazon Cognito_](https://aws.amazon.com/cognito/) through the provided [_Amazon Amplify_](https://aws.amazon.com/amplify/) javascript library. As an aside, the client application uses assets that are hosted using [AWS S3 Storage](https://aws.amazon.com/s3/). 
+
+The whole system is hosted using [_AWS Elastic Beanstalk_](https://aws.amazon.com/elasticbeanstalk/) and [_Amazon EC2_](https://aws.amazon.com/ec2/).
+
+The system also uses a custom Amazon Lambda function with a post-confirmation trigger for copying user-data (AWS-ID and username) to the system database. The trigger happens when a new user confirms their email through the provided sign-up system, simply sending a POST request with the data to the ```/users``` endpoint. 
+
+For an efficient development process, Continuous Deployment was implemented through [_AWS CodePipeline_](https://aws.amazon.com/codepipeline/).
+
+Furthermore, AWS hosts the [MongoDB](https://www.mongodb.com/) database for the server application. Due to technical difficulties the connection string to the  MongoDB instance is stored in plain text in the source code. However, the MongoDB instance only has the production server address whitelisted.  
```

Comment:
We should probably mention that the MongoDB instance is setup through MongoDB and not through AWS' database services. MongoDB then uses AWS, so we interface the database through MongoDB and not AWS.